src/IssueViz.Security.UnitTests/Hotspots/HotspotMatcherTests.cs

```diff
@@ -30,6 +36,188 @@ public class HotspotMatcherTests
     [TestMethod]
     public void MefCtor_CheckExports()
     {
-        MefTestHelpers.CheckTypeCanBeImported<NoOpHotspotMatcher, IHotspotMatcher>();
+        MefTestHelpers.CheckTypeCanBeImported<HotspotMatcher, IHotspotMatcher>();
+    }
+    
+    [TestMethod]
+    public void IsMatch_NoHashDifferentLineSameMessage_ReturnsTrue()
```

Comment:
It's difficult to tell whether all of the possible cases are covered or not from the test names.

Ideally most of the tests would be varying one value at a time and testing for both `true` and `false` cases e.g. this test would be stronger (and more isolated) if it had different inputs for the `message`, and tested the product code only returned `true` when the messages were identical and in the same case.

e.g.
```
    [TestMethod]
    [DataRow(null, null, false)] // null server has -> no match
    [DataRow(null, "any", false)] // null server hash -> no match
    [DataRow("", "", false)] // empty server -> no match
    [DataRow("", "any", false)] // empty server -> no match
    [DataRow("aaa", "AAA", false)] // different case - no match
    [DataRow("sameHash", "sameHash", true)]
    public void IsMatch_SameFileAndRuleId_VaryLineHashes(string serverHash, string localHash, bool shouldMatch)
    {
        // Different lines and messages.
        // RuleIds and paths do match.
        const string ruleId = "rule1";
        const string filePath = "A:\\ny\\p\\ath";
        const string serverPath = "p\\ath";

        var testSubject = CreateTestSubject();

        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message 1", localHash, filePath, 10),
        CreateServerHotspot(ruleId, "message 2", serverHash, serverPath, 20))
            .Should()
            .Be(shouldMatch);
    }

    [TestMethod]
    [DataRow(0, 1, false)]
    [DataRow(1, 0, false)]
    [DataRow(0, 0, true)]
    [DataRow(1, 1, true)]
    public void IsMatch_SameFileAndRuleId_VaryLineNumber(int serverLineNumber, int localLineNumber, bool shouldMatch)
    {
        // Different hashes and messages.
        // RuleIds and paths do match.
        const string ruleId = "rule1";
        const string filePath = "A:\\ny\\p\\ath";
        const string serverPath = "p\\ath";

        var testSubject = CreateTestSubject();

        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message 1", "localHash", filePath, serverLineNumber),
        CreateServerHotspot(ruleId, "message 2", "serverHash", serverPath, localLineNumber))
            .Should()
            .Be(shouldMatch);
    }

    [TestMethod]
    [DataRow(null, null, true)]
    [DataRow("", "", true)]
    [DataRow("aaa", "AAA", false)] // different case -> no match
    [DataRow("sameMessage", "sameMessage", true)]
    public void IsMatch_SameFileAndRuleId_VaryMessages(string serverMessage, string localMessage, bool shouldMatch)
    {
        // Different lines and hashes.
        // RuleIds and paths do match.
        const string ruleId = "rule1";
        const string filePath = "A:\\ny\\p\\ath";
        const string serverPath = "p\\ath";

        var testSubject = CreateTestSubject();

        testSubject.IsMatch(CreateLocalHotspot(ruleId, localMessage, "localHash", filePath, 10),
        CreateServerHotspot(ruleId, serverMessage, "serverHash", serverPath, 20))
            .Should()
            .Be(shouldMatch);
    }
```

I think those example tests plus
`IsMatch_DifferentRuleId_ReturnsFalse` and `IsMatch_IdenticalIssuesExceptFilePath_ReturnsBasedOnFilePathMatch` cover all of the cases and are easier to follow.